lib/backend/k8s/resources/networkpolicy.go

```diff
@@ -0,0 +1,295 @@
+	key := kvp.Key.(model.ResourceKey)
+	if strings.HasPrefix(key.Name, "knp.default.") {
+		// We don't support Create of a Kubernetes NetworkPolicy.
+		return nil, cerrors.ErrorOperationNotSupported{
+			Identifier: kvp.Key,
+			Operation:  "Create",
+		}
+	}
+	return c.crdClient.Create(ctx, kvp)
+}
+
+func (c *networkPolicyClient) Update(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
+	log.Debug("Received Update request on NetworkPolicy type")
+	key := kvp.Key.(model.ResourceKey)
+	if strings.HasPrefix(key.Name, "knp.default.") {
+		// We don't support Update of a Kubernetes NetworkPolicy.
+		return nil, cerrors.ErrorOperationNotSupported{
+			Identifier: kvp.Key,
+			Operation:  "Update",
+		}
+	}
+	return c.crdClient.Update(ctx, kvp)
+}
+
+func (c *networkPolicyClient) Apply(kvp *model.KVPair) (*model.KVPair, error) {
+	return nil, cerrors.ErrorOperationNotSupported{
+		Identifier: kvp.Key,
+		Operation:  "Apply",
+	}
+}
+
+func (c *networkPolicyClient) Delete(ctx context.Context, key model.Key, revision string) (*model.KVPair, error) {
+	log.Debug("Received Delete request on NetworkPolicy type")
+	k := key.(model.ResourceKey)
+	if strings.HasPrefix(k.Name, "knp.default.") {
+		// We don't support Delete of a Kubernetes NetworkPolicy.
+		return nil, cerrors.ErrorOperationNotSupported{
+			Identifier: key,
+			Operation:  "Delete",
+		}
+	}
+	return c.crdClient.Delete(ctx, key, revision)
+}
+
+func (c *networkPolicyClient) Get(ctx context.Context, key model.Key, revision string) (*model.KVPair, error) {
+	log.Debug("Received Get request on NetworkPolicy type")
+	k := key.(model.ResourceKey)
+	if k.Name == "" {
+		return nil, errors.New("Missing policy name")
+	}
+
+	// Check to see if this is backed by a NetworkPolicy.
+	if strings.HasPrefix(k.Name, "knp.default.") {
+		// Backed by a NetworkPolicy. Parse out the namespace / name.
+		namespace, policyName, err := c.converter.ParsePolicyNameNetworkPolicy(k.Name)
+		if err != nil {
+			return nil, cerrors.ErrorResourceDoesNotExist{Err: err, Identifier: k}
+		}
+
+		// Get the NetworkPolicy from the API and convert it.
+		networkPolicy := extensions.NetworkPolicy{}
+		err = c.clientSet.Extensions().RESTClient().
+			Get().
+			Resource("networkpolicies").
+			Namespace(namespace).
+			Name(policyName).
+			Timeout(10 * time.Second).
+			Do().Into(&networkPolicy)
+		if err != nil {
+			return nil, K8sErrorToCalico(err, k)
+		}
+		return c.converter.NetworkPolicyToPolicy(&networkPolicy)
+	} else {
+		return c.crdClient.Get(ctx, k, revision)
+	}
+}
+
+func (c *networkPolicyClient) List(ctx context.Context, list model.ListInterface, revision string) (*model.KVPairList, error) {
+	log.Debug("Received List request on NetworkPolicy type")
+	l := list.(model.ResourceListOptions)
+	if l.Name != "" {
+		// Exact lookup on a NetworkPolicy.
+		kvp, err := c.Get(ctx, model.ResourceKey{Name: l.Name, Kind: l.Kind}, revision)
+		if err != nil {
+			switch err.(type) {
```

Comment:
nit:  Given we are only testing a single type, perhaps `_, ok := err.(errors.ErrorResourceDoesNotExist)` is a little more succinct.